lib/new_relic/agent/configuration/default_source.rb

```diff
@@ -1463,6 +1463,15 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :allowed_from_server => false,
           :description => 'Controls auto-instrumentation of bunny at start-up. May be one of: `auto`, `prepend`, `chain`, `disabled`.'
         },
+        :'instrumentation.opensearch' => {
```

Comment:
The gem is called `opensearch-ruby`. The class is spelled `OpenSearch`, which is also the spelling used in the documentation. 

I'm torn between whether the lowercase version of the name should be `opensearch`, `open_search`, or `opensearch-ruby`.

Reply:
My order of preference is:

1. `opensearch-ruby` - match the gem name. I like this best.
2. `open_search` - given that the gem is a first party offering from the official OpenSearch project, converting the camel-cased "OpenSearch" into snake-cased "open_search" is reasonable.
3. `opensearch` - I want to not like this one, but they did name the gem 'opensearch-ruby' instead of 'open_search-ruby', so I suppose I can't argue with the lack of an underscore too much.

Reply:
Thank you! One thing on `opensearch-ruby`, I think we normally convert `-` to `_` for config options and directories. I'm not sure why, but for consistency with that process, we may want to use `_` if we go with the gem name.

Reply:
Ah, I see. So no matter what no approach will be perfect. Perhaps just merge this as-is then, as it makes plenty of sense!